guppylang/module.py

```diff
@@ -180,9 +181,18 @@ def compile(self) -> Hugr:
         )
         self._globals = self._globals.update_defs(type_defs)
 
+        # Collect auto-generated methods
+        generated: dict[DefId, RawDef] = {}
+        for defn in type_defs.values():
+            if isinstance(defn, CheckedStructDef):
+                for method_def in defn.generated_methods:
+                    generated[method_def.id] = method_def
+                    self._globals.impls.setdefault(defn.id, {})
```

Comment:
I think this wants to happen in the outer loop, so that the impls aren't set to `{}` each time a method is added

Reply:
`setdefault` only sets if the key doesn't exist yet

https://docs.python.org/3/library/stdtypes.html#dict.setdefault

Reply:
Okay, I think this still wants to be in the outer loop, but is ultimately harmless then